lib/model/model.go

```diff
@@ -1355,55 +1355,9 @@ func (m *Model) Request(deviceID protocol.DeviceID, folder, name string, offset
 		return protocol.ErrGeneric
 	}
 
-	if !scanner.Validate(buf, hash, weakHash) {
-		m.recheckFile(deviceID, folderFs, folder, name, int(offset)/len(buf), hash)
-		return protocol.ErrNoSuchFile
-	}
```

Comment:
I'd just remove `m.rcheckFile`, but keep validating and returning an error if it fails.